RecoTracker/FinalTrackSelectors/plugins/SingleLongTrackProducer.cc

```diff
@@ -108,21 +108,22 @@ void SingleLongTrackProducer::produce(edm::Event &iEvent, const edm::EventSetup
 
     TLorentzVector pTrack(track.px(), track.py(), track.pz(), track.pt());
 
-    // Long track needs to be close to a good muon
-    for (const auto &m : *muons) {
-      if (m.isTrackerMuon()) {
-        tMuon++;
-        reco::Track matchedTrack = *(m.innerTrack());
-        TLorentzVector pMatched(matchedTrack.px(), matchedTrack.py(), matchedTrack.pz(), matchedTrack.pt());
-        // match to general track in deltaR
-        double dr = pTrack.DeltaR(pMatched);
-        if (dr < dRmin)
-          dRmin = dr;
+    // Long track needs to be close to a good muon (only if requested)
+    if (matchInDr > 0.) {
+      for (const auto &m : *muons) {
+        if (m.isTrackerMuon()) {
+          tMuon++;
+          reco::Track matchedTrack = *(m.innerTrack());
+          TLorentzVector pMatched(matchedTrack.px(), matchedTrack.py(), matchedTrack.pz(), matchedTrack.pt());
+          // match to general track in deltaR
+          double dr = pTrack.DeltaR(pMatched);
```

Comment:
Using DeltaR2 could lead to some (tiny, here) perf improvement

Reply:
Thanks for the suggestion, implemented at https://github.com/cms-sw/cmssw/pull/43729/commits/3da22f135cede8946c5fb127de8f66bbd4905ba0